File: src/exemplos/05_EstruturasDeDados/01_Numeros/03-precisao_float.c
/**      @file: 04-precisao_float.c
 *     @author: Guilherme N. Ramos (gnramos@unb.br)
 * @disciplina: Algoritmos e Programação de Computadores
 *
 * Implementação de código para exemplificar o erro de
 * precisão com tipo float.
 *
 * Exemplo do livro A Whirlwind Tour of Python, de Jake VanderPlas
 * http://www.oreilly.com/programming/free/files/a-whirlwind-tour-of-python.pdf */

#include <stdio.h>

int main() {
    if (0.1 + 0.2 == 0.3)
        printf("0.1 + 0.2 == 0.3\n");
    else
        printf("0.1 + 0.2 != 0.3\n");

    printf("\n0.1 == %.17f\n", 0.1);
    printf("0.2 == %.17f\n", 0.2);
    printf("0.3 == %.17f\n", 0.3);

  return 0;
}

File: src/exemplos/05_EstruturasDeDados/01_Numeros/03-precisao_float.py
#  -*- coding: utf-8 -*-
#    @package: 03-precisao.py
#     @author: Guilherme N. Ramos (gnramos@unb.br)
# @disciplina: Algoritmos e Programação de Computadores
#
# Implementação de código para exemplificar o erro de precisão com ponto
# flutuante.
#
# Exemplo do livro A Whirlwind Tour of Python, de Jake VanderPlas
# http://www.oreilly.com/programming/free/files/a-whirlwind-tour-of-python.pdf

if 0.1 + 0.2 == 0.3:
    print('0.1 + 0.2 == 0.3')
else:
    print('0.1 + 0.2 != 0.3')

print('\n0.1 = {0:.17f}'.format(0.1))
print('0.2 = {0:.17f}'.format(0.2))
print('0.3 = {0:.17f}'.format(0.3))
